### PR TITLE
Using PER_JOB_HISTORY_DIR for APEL reporting (HTCONDOR-293)

### PR DESCRIPTION
**Pull request by markcoatsworth**
New clean pull request for these changes, including fixes from previous comments

### CODE REVIEW

File: contrib/apelscripts/condor_batch_blah.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        -format "%d|" JobStartDate \
+        -format "%d|" EnteredCurrentStatus \
+        -format "%d|" ResidentSetSize_RAW \
+        -format "%d|" ImageSize_RAW \
+        -format "%d|" RequestCpus \
+        "${CONDOR_Q_EXTRA_ARGS[@]}"`
+    OUTPUT_FILE="$OUTPUT_DIR/batch-${OUTPUT_DATETIME}-$(hostname -s)"
+    echo $batch_record >> $OUTPUT_FILE
+
+    blah_record=`TZ=GMT condor_q -job $file \
```
Comment (brianhlin) [inline contrib/apelscripts/condor_batch_blah.sh:71]:
```suggestion
    blah_record=`TZ=GMT condor_q -job "$file" \
```

File: contrib/apelscripts/condor_batch_blah.sh [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    fi
+
+    # Check if $file is a valid history file by looking for a ClusterId value
+    # If not found, assume the file is invalid and move to quarantine folder
+    if [[ -z `condor_q -job $file -format \"%s\" ClusterId` ]]; then
+        mv $file $QUARANTINE_DIR
+        continue
+    fi
+
+    batch_record=`TZ=GMT condor_q -job $file \
```
Comment (brianhlin) [inline contrib/apelscripts/condor_batch_blah.sh:55]:
```suggestion
    batch_record=`TZ=GMT condor_q -job "$file" \
```

File: contrib/apelscripts/condor_batch_blah.sh [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        -format "%s" ClusterId \
+        -format ".%s_${BATCH_HOST}|" ProcId \
```
Comment (brianhlin) [inline contrib/apelscripts/condor_batch_blah.sh:57]:
```suggestion
    -format "%s|" GlobalJobId \
```

This was changed in #426

File: contrib/apelscripts/condor_batch_blah.sh [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    echo $batch_record >> $OUTPUT_FILE
+
+    blah_record=`TZ=GMT condor_q -job $file \
+        -format "\"timestamp=%s\" " 'formatTime(EnteredCurrentStatus, "%Y-%m-%d %H:%M:%S")' \
+        -format "\"userDN=%s\" " x509userproxysubject \
+        -format "\"userFQAN=%s\" " x509UserProxyFirstFQAN \
+        -format "\"ceID=${CE_ID}\" " EMPTY \
+        -format "\"jobID=%v_${CE_HOST}\" " RoutedFromJobId \
+        -format "\"lrmsID=%v" ClusterId \
+        -format ".%v_${BATCH_HOST}\" " ProcId \
```
Comment (brianhlin) [inline contrib/apelscripts/condor_batch_blah.sh:78]:
```suggestion
        -format "\"lrmsID=%s\" " GlobalJobId \
```

The matching change from #426

**Review by brianhlin — CHANGES_REQUESTED** [review]
It looks like this commit accidentally stomps on changes related to #426. Also a few other suggestions.

**Comment by markcoatsworth** [comment]
@brianhlin I just pushed an update which addresses all your previous comments. I also changed the Is-History-File-Valid check to look for a GlobalJobId instead of ClusterId since we're now using the former as the unique job identifier.

**Review by brianhlin — APPROVED** [review]
LGTM! Overriding unrelated osg-test failures

**Pull request merged**